### PR TITLE
(MAINT) Rubocop updates for spec

### PR DESCRIPTION
**Pull request by chelnak**
This PR is a continuation pf #54. It contains rubocop fixes for all code in `spec`.

### CODE REVIEW

File: spec/puppet-lint/plugins/check_resources/file_mode_spec.rb [hunk 1]
```diff
@@ -168,86 +168,86 @@
...
         expect(problems).to have(1).problem
       end
 
-      it 'should fix the manifest' do
+      it 'fixes the manifest' do
         expect(problems).to contain_fixed(msg).on_line(1).in_column(23)
       end
 
-      it 'should zero pad the file mode' do
+      it 'zeroes pad the file mode' do
```
Comment (GSPatton) [inline spec/puppet-lint/plugins/check_resources/file_mode_spec.rb:190]: should be "zero pads"? Or keep as is?

File: spec/puppet-lint/plugins/check_resources/file_mode_spec.rb [hunk 2]
```diff
@@ -168,86 +168,86 @@
...
         expect(problems).to have(1).problem
       end
 
-      it 'should fix the manifest' do
+      it 'fixes the manifest' do
         expect(problems).to contain_fixed(msg).on_line(1).in_column(25)
       end
 
-      it 'should zero pad the concat mode' do
+      it 'zeroes pad the concat mode' do
```
Comment (GSPatton) [inline spec/puppet-lint/plugins/check_resources/file_mode_spec.rb:222]: same as above - only one other instance of this

File: spec/puppet-lint/plugins/check_resources/unquoted_file_mode_spec.rb [hunk 3]
```diff
@@ -64,54 +64,54 @@
...
         expect(problems).to have(1).problem
       end
 
-      it 'should fix the manifest' do
+      it 'fixes the manifest' do
         expect(problems).to contain_fixed(msg).on_line(1).in_column(23)
       end
 
-      it 'should single quote the file mode' do
+      it 'singles quote the file mode' do
```
Comment (GSPatton) [inline spec/puppet-lint/plugins/check_resources/unquoted_file_mode_spec.rb:86]: maybe keep the same

File: spec/puppet-lint/plugins/check_resources/unquoted_file_mode_spec.rb [hunk 4]
```diff
@@ -64,54 +64,54 @@
...
         expect(problems).to have(1).problem
       end
 
-      it 'should fix the manifest' do
+      it 'fixes the manifest' do
         expect(problems).to contain_fixed(msg).on_line(1).in_column(25)
       end
 
-      it 'should single quote the file mode' do
+      it 'singles quote the file mode' do
```
Comment (GSPatton) [inline spec/puppet-lint/plugins/check_resources/unquoted_file_mode_spec.rb:102]: same as above - 5 instances of this in total

File: spec/puppet-lint/plugins/check_strings/only_variable_string_spec.rb [hunk 5]
```diff
@@ -50,65 +50,65 @@
...
         expect(problems).to have(1).problem
       end
 
-      it 'should fix the manifest' do
+      it 'fixes the manifest' do
         expect(problems).to contain_fixed(msg).on_line(1).in_column(4)
       end
 
-      it 'should unquoted the variable' do
+      it 'unquoteds the variable' do
```
Comment (GSPatton) [inline spec/puppet-lint/plugins/check_strings/only_variable_string_spec.rb:95]: unquotes

File: spec/puppet-lint/checks_spec.rb [hunk 6]
```diff
@@ -74,12 +74,12 @@
     let(:data) { "notify { 'test': }" }
     let(:enabled_checks) { [] }
 
-    before do
-      allow(instance).to receive(:enabled_checks).and_return(enabled_checks)
+    before(:each) do
+      allow(instance).to receive(:enabled_checks).and_return(enabled_checks) # rubocop: disable RSpec/SubjectStub
```
[the remark this answers is not in the document]
Comment (chelnak) [reply spec/puppet-lint/checks_spec.rb:78]: Yeah, comments like this disable the linting rule for line given line.

**Comment by chelnak** [comment]
@GSPatton With regards to the example names, this is enforced by [this rule](https://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/ExampleWording). Check it out for some further examples.

File: spec/puppet-lint/plugins/check_classes/inherits_across_namespaces_spec.rb [hunk 7]
```diff
@@ -6,27 +6,27 @@
...
+    it 'does not detect any problems' do
       expect(problems).to have(0).problems
     end
   end
 
   context 'class inheriting from another module namespace' do
     let(:code) { 'class foo::bar inherits baz { }' }
 
-    it 'should only detect a single problem' do
+    it 'onlies detect a single problem' do
```
Comment (GSPatton) [inline spec/puppet-lint/plugins/check_classes/inherits_across_namespaces_spec.rb:25]: there are still many occurrences of "onlies" - 99 of them I think... would it be an issue to fix all of them?

File: spec/puppet-lint/plugins/check_resources/file_mode_spec.rb [hunk 8]
```diff
@@ -281,17 +281,17 @@
...
 
-      it 'should fix 3 problems' do
+      it 'fixes 3 problems' do
         expect(problems).to contain_fixed(msg).on_line(4).in_column(23)
         expect(problems).to contain_fixed(msg).on_line(6).in_column(23)
         expect(problems).to contain_fixed(msg).on_line(8).in_column(23)
       end
 
-      it 'should zero pad the file modes and change them to strings' do
+      it 'zeroes pad the file modes and change them to strings' do
```
Comment (GSPatton) [inline spec/puppet-lint/plugins/check_resources/file_mode_spec.rb:294]: one more occurrence of "zeroes pad"

File: spec/puppet-lint/plugins/check_classes/inherits_across_namespaces_spec.rb (hunk 7 above)
Comment (chelnak) [reply spec/puppet-lint/plugins/check_classes/inherits_across_namespaces_spec.rb]: 🤣

**Review by GSPatton — APPROVED** [review]
looks good

**Pull request merged**